server/models/plugins/basePlugin.js

```diff
@@ -0,0 +1,103 @@
+const mongoose = require('mongoose')
+
+const BasePluginSchema = new mongoose.Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+    },
+    resourceId: {
+      type: Schema.Types.ObjectId,
+    },
+    description: {
+      type: String,
+      required: true,
+    },
+  },
+  { timestamps: true }
+)
+
+BasePluginSchema.methods.get = function () {
+  return await this.findOne({ _id: this._id })
+}
+
+BasePluginSchema.methods.post = async () => {}
+
+BasePluginSchema.methods.update = async (objectInfo) => {
```

Comment:
Can we double check if BasePluginSchema has instance methods, from the issue I read two static methods

Reply:
This is fine, it's good for testing the plugin controllers and routes 🙂